src/window.rs

```diff
@@ -95,6 +114,120 @@ fn main() {
     }
 }
 
+struct View {
+    pub cell_width: f64,
+    pub base_cell_width: f64,
+
+    pub y: usize,
+    pub x: usize,
+
+    pub precise_y: f64,
+    pub precise_x: f64,
+
+    pub capture_cursor: bool,
+
+    pub cells_on_width: usize,
+    pub cells_on_height: usize,
+
+    pub board_width: usize,
+    pub board_height: usize,
+
+    pub window_width: u32,
+    pub window_height: u32,
+}
+
+impl View {
+    fn toggle_capture_cursor(&mut self) {
+        self.capture_cursor = !self.capture_cursor;
+    }
+
+    fn from_config(config: &Config) -> Self {
+        let board_width = config.width as usize;
+        let board_height = config.height as usize;
+
+        let base_cell_width: f64 = config.cell_width.into();
+        let cell_width = base_cell_width.clone();
+
+        let window_width = if board_width as f64 * cell_width > 600.0 {600} else {(board_width as f64 * cell_width) as u32};
+        let window_height = if board_height as f64 * cell_width > 600.0 {600} else {(board_height as f64 * cell_width) as u32};
```

Comment:
These `600`s shouldn't be constants, but rather be read from the command line arguments.

Reply:
Yeah, that was temporary and should indeed be removed before this gets merged.

Reply:
I am currently looking at [glutin](https://docs.rs/glutin/0.14.0/glutin/index.html) if I can get the screen size, this will allow me to limit a window to the screen size. I don't think the window size should be read from the command line since I see little use in it and determining it can be automated.